File: apex_habitat/salis/report/movement_cost_recovery_register/movement_cost_recovery_register.js
// Copyright (c) 2026, AFMCO and contributors
frappe.query_reports["Movement Cost Recovery Register"] = {
	filters: [
		{
			fieldname: "company",
			label: __("Company"),
			fieldtype: "Link",
			options: "Company",
			default: frappe.defaults.get_user_default("Company"),
		},
		{
			fieldname: "cost_center",
			label: __("Cost Center"),
			fieldtype: "Link",
			options: "Cost Center",
		},
		{
			fieldname: "recovery_type",
			label: __("Recovery Type"),
			fieldtype: "Select",
			options: ["", "Vehicle Damage", "Fuel Misuse", "Custody Loss", "Fine / Violation", "Other"].join("\n"),
		},
		{
			fieldname: "status",
			label: __("Status"),
			fieldtype: "Select",
			options: ["", "Open", "Acknowledged", "Approved", "Recovered", "Waived", "Rejected"].join("\n"),
		},
		{
			fieldname: "vehicle",
			label: __("Vehicle"),
			fieldtype: "Link",
			options: "Salis Vehicle",
		},
		{
			fieldname: "driver",
			label: __("Driver"),
			fieldtype: "Link",
			options: "Salis Driver",
		},
		{
			fieldname: "employee",
			label: __("Employee"),
			fieldtype: "Link",
			options: "Employee",
		},
		{
			fieldname: "from_date",
			label: __("From Date"),
			fieldtype: "Date",
		},
		{
			fieldname: "to_date",
			label: __("To Date"),
			fieldtype: "Date",
		},
	],
};

File: apex_habitat/salis/report/movement_cost_recovery_register/movement_cost_recovery_register.py
# Copyright (c) 2026, AFMCO and contributors
# For license information, please see license.txt

import frappe
from frappe import _


def execute(filters=None):
    columns = [
        {"label": _("Recovery"), "fieldname": "name", "fieldtype": "Link", "options": "Movement Cost Recovery", "width": 180},
        {"label": _("Recovery Type"), "fieldname": "recovery_type", "fieldtype": "Data", "width": 150},
        {"label": _("Vehicle"), "fieldname": "vehicle", "fieldtype": "Link", "options": "Salis Vehicle", "width": 150},
        {"label": _("Driver"), "fieldname": "driver", "fieldtype": "Link", "options": "Salis Driver", "width": 150},
        {"label": _("Employee"), "fieldname": "employee", "fieldtype": "Link", "options": "Employee", "width": 150},
        {"label": _("Amount"), "fieldname": "amount", "fieldtype": "Currency", "width": 130},
        {"label": _("Status"), "fieldname": "status", "fieldtype": "Data", "width": 130},
        {"label": _("Payment Request"), "fieldname": "payment_request", "fieldtype": "Link", "options": "Salis Payment Request", "width": 180},
    ]

    query_filters = {}
    if filters:
        for field in ("status", "recovery_type", "vehicle", "driver", "employee", "company", "cost_center"):
            if filters.get(field):
                query_filters[field] = filters[field]
        if filters.get("from_date") and filters.get("to_date"):
            query_filters["creation"] = ["between", [filters["from_date"], filters["to_date"]]]
        elif filters.get("from_date"):
            query_filters["creation"] = [">=", filters["from_date"]]
        elif filters.get("to_date"):
            query_filters["creation"] = ["<=", filters["to_date"]]

    data = frappe.get_all(
        "Movement Cost Recovery",
        filters=query_filters,
        fields=[
            "name",
            "recovery_type",
            "vehicle",
            "driver",
            "employee",
            "amount",
            "status",
            "payment_request",
            "creation",
        ],
        order_by="creation desc",
    )

    return columns, data
